### ISSUE TITLE
I want to know the test process

### ISSUE THREAD

**Issue by LYG96** [dimitymiller/cac-openset#6]
I want to know which part of the code is represented by equation 8 and equation 9 in the test part of the pape。thank you！

**Comment by dimitymiller** [comment]
The calculation of the rejection score (equation 8) in utils.py, inside the gather_outputs function and on lines 168-170:
`softmin = softmax(-distances)`
`invScores = 1-softmin`
`scores = distances*invScores`

The rejection of inputs based on their rejection score (equation 9) is in metrics.py, inside the auroc function and on line 22:
`fpr, tpr, thresholds = sklearn.metrics.roc_curve(labels, allData, pos_label = in_low)`
It relies on the sklearn.metrics.roc_curve function which rejects inputs at each of the thresholds that is returned, and calculates the FPR and TPR at that threshold.
If you want to only reject an input given a threshold without calculating FPR and TPR you would need to implement this.
